applications/GeoMechanicsApplication/tests/cpp_tests/test_transient_thermal_element.cpp

```diff
@@ -0,0 +1,428 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Gennady Markelov
+//
+
+#include "containers/model.h"
+#include "custom_elements/transient_thermal_element.h"
+#include "geo_mechanics_application_variables.h"
+#include "testing/testing.h"
+
+using namespace Kratos;
+
+namespace Kratos::Testing {
+
+void TestThermalElement(ModelPart& rModelPart)
+{
+    Element::Pointer p_element = rModelPart.pGetElement(1);
+
+    for (unsigned int i = 0; i < rModelPart.NumberOfNodes(); i++) {
+        p_element->GetGeometry()[i].AddDof(TEMPERATURE);
+    }
+
+    const ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+    // p_element->Check(r_current_process_info);
+    // Check requires a solution?
+
+    Element::DofsVectorType ElementalDofList;
+    p_element->GetDofList(ElementalDofList, r_current_process_info);
+
+    for (unsigned int i = 0; i < ElementalDofList.size(); i++) {
+        ElementalDofList[i]->SetEquationId(i);
+    }
+
+    Element::EquationIdVectorType EquationIdVector;
+    p_element->EquationIdVector(EquationIdVector, r_current_process_info);
+
+    // Check the EquationIdVector values
+    for (unsigned int i = 0; i < EquationIdVector.size(); i++) {
+        KRATOS_EXPECT_TRUE(EquationIdVector[i] == i);
```

Comment:
We are missing some more asserts and checks on the element, but since we want to start testing with the integration tests as soon as possible, that could be a separate PR